internal/extender/resource.go

```diff
@@ -439,6 +441,14 @@ func (s *SparkSchedulerExtender) findUnboundReservations(ctx context.Context, ex
 	return relocatableReservations, successRescheduled, nil
 }
 
+func isPodTerminated(pod *v1.Pod) bool {
+	allTerminated := false
```

Comment:
I think this should be initialised with `len(pod.Status.ContainerStatuses) > 0`. We can also early return if the container statuses is empty and set allTerminated to true and fold from there